File: src/test/java/ukalus/path/astar/Node2DTest.java
/*
 * Created on Feb 11, 2005
 *
 */
package ukalus.path.astar;

import ukalus.level.Level;
import ukalus.math.Vector2D;

import java.util.Collection;

import junit.framework.TestCase;

import static java.lang.Boolean.*;

/**
 * @author gremopm
 * 
 */
public class Node2DTest extends TestCase {

  public void testGetSuccessorsFromMiddle() {
    Level<Boolean> map = new Level<>(new Boolean[][]{
        {TRUE, TRUE, TRUE},
        {TRUE, TRUE, TRUE},
        {TRUE, TRUE, TRUE}});
    Node2D parent = new Node2D(map, Vector2D.Companion.get(1, 0), null);
    Node2D current = new Node2D(map, Vector2D.Companion.get(1, 1), parent);
    Collection<Node> successors = current.getSuccessors();
    assertEquals(3, successors.size());
  }

  public void testGetSuccessorsFromCorner() {
    Level<Boolean> map = new Level<>(new Boolean[][]{
        {TRUE, TRUE, TRUE},
        {TRUE, TRUE, TRUE},
        {TRUE, TRUE, TRUE}});
    Node2D parent = new Node2D(map, Vector2D.Companion.get(1, 0), null);
    Node2D current = new Node2D(map, Vector2D.Companion.get(0, 0), parent);
    Collection<Node> successors = current.getSuccessors();
    assertEquals(1, successors.size());
  }

  public void testGetSuccessorsFromEdge() {
    Level<Boolean> map = new Level<>(new Boolean[][]{
        {TRUE, TRUE, TRUE},
        {TRUE, TRUE, TRUE},
        {TRUE, TRUE, TRUE}});
    Node2D parent = new Node2D(map, Vector2D.Companion.get(0, 0), null);
    Node2D current = new Node2D(map, Vector2D.Companion.get(0, 1), parent);
    Collection<Node> successors = current.getSuccessors();
    assertEquals(2, successors.size());
  }

  public void testGetSuccessorsOriginInMiddle() {
    Level<Boolean> map = new Level<>(new Boolean[][]{
        {TRUE, TRUE, TRUE},
        {TRUE, TRUE, TRUE},
        {TRUE, TRUE, TRUE}});
    Node2D current = new Node2D(map, Vector2D.Companion.get(1, 1), null);
    Collection<Node> successors = current.getSuccessors();
    assertEquals(4, successors.size());
  }
}

File: src/test/java/ukalus/path/astar/Node2DTest.kt
/*
 * Created on Feb 11, 2005
 *
 */
package ukalus.path.astar

import ukalus.level.Level
import ukalus.math.Vector2D

import junit.framework.TestCase

import java.lang.Boolean.*

/**
 * @author gremopm
 */
class Node2DTest : TestCase() {

    fun testGetSuccessorsFromMiddle() {
        val map = Level(arrayOf(arrayOf(TRUE, TRUE, TRUE), arrayOf(TRUE, TRUE, TRUE), arrayOf(TRUE, TRUE, TRUE)))
        val parent = Node2D(map, Vector2D(1, 0), null)
        val current = Node2D(map, Vector2D(1, 1), parent)
        val successors = current.successors
        TestCase.assertEquals(3, successors.size)
    }

    fun testGetSuccessorsFromCorner() {
        val map = Level(arrayOf(arrayOf(TRUE, TRUE, TRUE), arrayOf(TRUE, TRUE, TRUE), arrayOf(TRUE, TRUE, TRUE)))
        val parent = Node2D(map, Vector2D(1, 0), null)
        val current = Node2D(map, Vector2D(0, 0), parent)
        val successors = current.successors
        TestCase.assertEquals(1, successors.size)
    }

    fun testGetSuccessorsFromEdge() {
        val map = Level(arrayOf(arrayOf(TRUE, TRUE, TRUE), arrayOf(TRUE, TRUE, TRUE), arrayOf(TRUE, TRUE, TRUE)))
        val parent = Node2D(map, Vector2D(0, 0), null)
        val current = Node2D(map, Vector2D(0, 1), parent)
        val successors = current.successors
        TestCase.assertEquals(2, successors.size)
    }

    fun testGetSuccessorsOriginInMiddle() {
        val map = Level(arrayOf(arrayOf(TRUE, TRUE, TRUE), arrayOf(TRUE, TRUE, TRUE), arrayOf(TRUE, TRUE, TRUE)))
        val current = Node2D(map, Vector2D(1, 1), null)
        val successors = current.successors
        TestCase.assertEquals(4, successors.size)
    }
}